src/main/scala/org/tensorframes/impl/datatypes.scala

```diff
@@ -488,4 +558,69 @@ private[impl] object LongOperations extends ScalarTypeOperation[Long] with Loggi
     logTrace(s"Extracted from buffer: ${res.toSeq}")
     res
   }
-}
\ No newline at end of file
+
+  override def tag: Option[TypeTag[_]] = Option(implicitly[TypeTag[Long]])
+
+  override def ev = ClassTag.Long
+}
+
+// ********** STRING *********
+// This is actually byte arrays, which corresponds to the 'binary' type in Spark.
```

Comment:
The `String` type in Spark is different: it corresponds to the UTF-8 encoded representation of textual data, and it would not be accepted. The reason for that is that you would need to specify the encoding (UTF-8, UTF-16, etc.) in order to convert it to a byte array. In order to bypass this problem, tensorflow only accepts byte arrays (which they call 'strings').

Under the hood, though, Spark passes byte arrays to this function (see `appendRaw` below)